[TorchToLinalg] Implement lowering of torch.aten.rrelu_with_noise and torch.aten.rrelu_with_noise_backward ops (fix)

bosko-syrmia opened the pull request:
As title. @vivekkhandelwal1 This PR finalizes and resolves issues from #3645. Updated `xfail_set` and cherry-picked previous implementation.

vivekkhandelwal1 reviewed (changes requested):
Just a small change, and the rest looks good.

vivekkhandelwal1 reviewed (approved):
LGTM

vivekkhandelwal1 commented on lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:
```diff
@@ -3588,6 +3641,82 @@ class DecomposeAtenRreluOp : public OpRewritePattern<AtenRreluOp> {
+
+    Value scaledSelf;
+    if (training) {
+      scaledSelf = rewriter.create<AtenMulTensorOp>(loc, resType, self, alpha);
+      auto boolResType = resType.getWithSizesAndDtype(resType.getSizes(),
+                                                      rewriter.getI1Type());
+      Value oneTensor =
+          createRank0Tensor(rewriter, loc, resType, constantOneFloat);
+      Value not_positive = rewriter.create<AtenLtScalarOp>(
+          loc, boolResType, self, constantZeroFloat);
+      noise = rewriter.create<AtenWhereSelfOp>(loc, resType, not_positive,
+                                               alpha, oneTensor);
+    } else {
+      scaledSelf = rewriter.create<AtenMulScalarOp>(loc, resType, self, alpha);
+    }
```
Sorry, I missed that difference.

zjgarvey commented:
Hey @vivekkhandelwal1 this seems to be the commit that caused the CI to break on nightly.

bosko-syrmia commented:
It seems that some tests started failing on later nightly builds, after I committed last changes. So I made this PR #3835 that addresses those issues.
